### ISSUE TITLE
Resolution & vision issues , Tricks & Tips

### ISSUE THREAD

**Issue by ODEH67** [LukeRoss00/gta5-real-mod#211]
This thread is to discuss The Resolution & vision issues u could face , and the suggested Tricks & Tips to solve or to enhance it

**Comment by ODEH67** [comment]
here I would like to share my Graphics card sittings (Nvidia settings) in order to see If i missed something to get the best quality and performance as possible in the Game, specially it offers some features for virtual reality...

**My Setup:**

**- GTA V steam version **

**- Windows 10 pro**

**- MB: Asus TUF GAMING X570-PLUS,**

**- AMD Ryzen 5 3600 WATER COOLED,**

**- (2 x 8 GB) DDR4-3200,**

**- Asus RTX 2080 super OC 8G extreme**

**- Nvidia Driver version : 446.14**

**- Headset: HP Reverb pro**

**- VR MOD release 5.0** 

**- Using Low sittings option from the mod and leave it on the mod Stock sittings inside the game**

![Screenshot (60)](https://user-images.githubusercontent.com/63449828/87290587-76ace100-c4fe-11ea-8c75-95898558ed5c.png)


![Screenshot (67)_LI](https://user-images.githubusercontent.com/63449828/87290593-79a7d180-c4fe-11ea-94fd-4c71f69e1307.jpg)

what should I change there specially the last two options which it is about VR settings?

**Comment by LukeRoss00** [comment]
Sad that no other users contributed to this thread, especially seeing as you have been very active on GitHub to help others.

I see that you are using Low settings beefed up with extremely high resolution (1800x1800, which translates to a 3600x3600 internal buffer with 2x scaling). That will make the game look sharp but IMO rather bleak. I would advise you to try the opposite, since you have a very powerful video card: start from High settings; set a base resolution of 1200x1200 (which will give you a 3000x3000 internal buffer with 2.5x scaling); get rid of the sparkly jaggies with MSAA 2x + TXAA; gain back some of the lost frame rate reducing PostFX to Normal. For additional awesomeness, you can also install M.V.G.A. to make the graphics almost photorealistic. With this setup, you will probably have less fluidity in terms of frame rate, but the game will look gorgeous.

The headset and display options seem correct to me. The 3 extra display are not the headset internal screen but virtual displays (see [here](https://www.reddit.com/r/WindowsMR/comments/hvbq9w/for_the_love_of_god_let_us_disable_the_virtual/) for details). The two Virtual Reality-related options in the NVIDIA control panel have no effect on the mod and you can leave them to the values you showed.

**Comment by ODEH67** [comment]
thanks for replying , I know it seems more look "Hit and run" thing , but I don't mind it actually specially if our Tips really help to solve their problems, which is very difficult to say when they mostly don't give their feedback so that would be helpful for the others who could face those problems.. and that is what I find it unfortunate.

about my experience with the settings i'm actually working on it by trying every possible scenario to achieve the best result,,,
(I had to do it long time ago but since a lot is going on I don't have enough time to actually do it continually:/ )
but story short about this particular resolution-choice; if I don't set the highest possible resolution with 2x or 2,5x times Up-scaling, the game in the headset will look so fussy specially the edges of the objects (trees, buildings , Road barrier/fence and so on) by  avoiding to use the FXAA, MSAA OR TXAA making the view more blurry which for me personally is not preferred.. 
choosing the low setting for the mod is in purpose to make less pressure on the GPU and letting it only work on the up-scaling  to gain more smoothness plus sharpness.. but still as u said to get that sharpness its on the other hand will not be that smooth ..

> start from High settings; gain back some of the lost frame rate reducing PostFX to Normal

I didn't try to use the High settings at all in my Test assuming that the lower u go with the settings the better smoothness u get.. but I would give that a shot 👍

> For additional awesomeness, you can also install M.V.G.A. to make the graphics almost photorealistic. With this setup, you will probably have less fluidity in terms of frame rate, but the game will look gorgeous

I already have the redux mod which is really breath-taking,the M.V.G.A uses the same method of installing which makes me wondering y would the VR mod works with it and not with the redux mod? on the **monitor** while using the VR mod the redux works and appears as it while playing in standard (with no VR) but it does not appear in the headset  :/


> The headset and display options seem correct to me. The 3 extra display are not the headset internal screen but virtual displays (see [here](https://www.reddit.com/r/WindowsMR/comments/hvbq9w/for_the_love_of_god_let_us_disable_the_virtual/) for details). The two Virtual Reality-related options in the NVIDIA control panel have no effect on the mod and you can leave them to the values you showed.

well that explains y changing the Monitor in the in-game settings does not make any change .. but that was something that I was thinking of for a while >>> does the Game-view on the Monitor while using the VR mod need to be buffered as well as on the headset or is it just streaming as a video? I mean does the gpu in this case needs to project on 3x displays (two times on headset and one time on the monitor)? since the game is not designed for VR , it could be that the view on the monitor takes some work from the GPU..

**Comment by LukeRoss00** [comment]
See [here](https://github.com/LukeRoss00/gta5-real-mod#more-advanced-graphics-tweaking) for my thoughts about M.V.G.A., Redux, etc. and for tips about how to install them so that they work in VR (basically you have to skip the ReShade installation and cannot use any ReShade presets).

The game window on the monitor just displays a copy of one eye buffer (the left eye by default), so no, it won't use any appreciable amount of GPU resources and certainly it won't draw everything again :-D

**Comment by ODEH67** [comment]
Hello @LukeRoss00,   well, it seems weird that almost everyone submit the same issue as a new issue although this thread made specially for those discussions... Vision issues in this case ... if an issue mentioned here once it will get solved and stay open for others who would face the same thing ... so before they submit the new issue as a new issue , they would find their answer... 
so for example; if i am a cosmos user with an vision issue , and someone gave me a tip for help regardless if he is an cosmos user too or not , if it worked for me , my feedback is like an approve that this tip is the solution for this particular issue and it works... but I dont know if u think maybe changing the Title would be more easier to understand what is it for? ...

> start from High settings; set a base resolution , gain back some of the lost frame rate reducing PostFX to Normal

btw I followed your suggestion about that and actually i noticed some better performance , I believe since the high settings where provided with the mod directly , so going slightly low in high settings better than going massively high in the low settings 👍

**Comment by LukeRoss00** [comment]
The OpenXR runtime provided by Oculus still has a few bugs to iron out, I'm talking with them but they move at glacial speed

**Comment by d0x360** [comment]
If you could add TAA injection to the mod it would give a huge performance and image quality boost.  There is a TAA injection mod on GitHub for alien isolation and it's amazing especially when playing alien isolation in vr.  It removes 100% of the shimmer and jaggies and doesn't cause ghosting.

**Comment by LukeRoss00** [comment]
@d0x360 I fail to see why you believe TAA would give a "huge performance boost" to the mod. Despite being relatively cheap, TAA still involves applying several postprocessing steps to the high-res eye buffers, plus keeping extra copies around and sampling from the depth buffer, so it would certainly diminish performance and not boost it. As for image quality, I must say that I'm not a fan of how TAA looks, especially in VR. Fallout 4 had a good, native implementation, and all people were ever discussing in forums was how to get rid of the horrible blur that TAA introduced.

**Comment by d0x360** [comment]
I say it because I've seen it work perfectly and it cost zero performance.
I gave an example of a game that wasn't meant for VR and I gave an example
of a TAA injection method that is on GitHub and it's literally the only
thing I've ever seen that was able to get rid of jaggies in alien
isolation.  Even running that game at 8k then downsampling to 4k doesn't
remove jaggies on alpha effects.

Let's say that performance cost isn't free...at worst is 2fps.

I also am not a fan of most implementations of TAA but again I've given an
example that you can try right now that works amazingly well and doesn't
hit you with a performance penalty.

The reason I brought it up is because none of gta5's built in AA options
really work except MSAA but MSAA has a massive performance penalty
especially compared to basically everything else.  It even has a higher
penalty per pixel than super sampling.

**Comment by LukeRoss00** [comment]
I just watched a video comparison of Alien: Isolation with TAA on and off and I agree that the implementation looks very good, at least judging from YouTube. I'll try to find some time to check it out in VR during the next few weeks.

I'm quite dubious whether the same results can be obtained with GTA V, though. If it were that simple, considering that GTA V is one of the most modded games of all time, and that as you mentioned all forms of antialiasing native to the game more or less suck for various reasons, at this point somebody would have tried to implement TAA in a GTA V mod too. There's probably something in the way the renderer is coded that prevents it from working. That, or nobody has ever thought of it, which strikes me as rather unlikely ;-)

**Comment by d0x360** [comment]
Nothing in the rendering pipeline would prevent it from working and nobody
has likely made an injection tool (as far as we know) because technically
the game already has a form of temporal anti aliasing but it was designed
for the xbox 360 and ps3.  They never updated it for the ps4, xbox one and
pc release although I would expect they will update it when they release it
again on xbox series x and ps5...but will that update hit PC if they even
update their AA methods?

There are multiple forms of temporal anti aliasing and outside of that it
also depends on how many samples are done.

I don't know the exact form used in the alien isolation injector but I do
know it works amazingly well.  Thankfully it's right here on GitHub and
should be able to be made to work with any other game.

**Comment by ODEH67** [comment]
hello, thanks for the suggestion @d0x360 , can u give the link for the TAA Injection so I can try that out? the Reverb G2 is coming soon in couple of two weeks, so it will be interesting to see if there will be any improvements specially on the software-side which it comes with any new hardware releases.

**Comment by d0x360** [comment]
Here's the reddit thread created by the injectors author.  It has links to the injector but I figured this would be better because you could send him a message if needed.

https://www.reddit.com/r/alienisolation/comments/508aet/alias_isolation_an_antialiasing_mod/

**Comment by ODEH67** [comment]
thanx that looks interesting 👍 , I think it would take some time to go through first to see if it really would work in GTA V beside the R.E.A.L mod..  
> I'll try to find some time to check it out in VR during the next few weeks
@LukeRoss00  I would wait for your opinion on that since it seems a **Pro-level** thing 😅

**Comment by d0x360** [comment]
It works on top of the vr mod and other mods in alien isolation.  I don't
see why it wouldn't work with anything, it's just a matter of changing what
executable it's looking for.

**Comment by LukeRoss00** [comment]
> I don't see why it wouldn't work with anything, it's just a matter of changing what executable it's looking for.

That's not how these things work. The TAA injection mod is specific to the Alien: Isolation game and its renderer, it would never work on GTA V without being rewritten from scratch. The two games are not even based on the same engine. It's certainly not "just a matter of changing what executable it's looking for" LOL

**Comment by d0x360** [comment]
Actuallly it is that simple.  It works the same way as reshade does.  You
tell reshade what game to hook into, it does the rest because it was coded
that way.

This TAA mod works the same way and if you don't believe me then ask the
author.  I have.

It doesn't matter what renderer it's using because both games run on dx11
and this is temporal aa.  So the AA pass is AFTER the final render pass is
complete.

I've worked in game development, I don't need a lesson nor do I need to be
spoken down to like some 13 year old moron who thinks they know what's what.

You have been negative and now insulting about this suggestion from moment
1.  If you didn't want to look into it then you shouldn't have asked for
more info.

I mean jesus you couldn't have even been bothered to do a Google search for
it.  Google "alien isolation TAA" and it comes up as the first result!

**Comment by LukeRoss00** [comment]
By all means, since you've worked in game development feel free to adapt the TAA injection mod to GTA V, and then if you're so inclined publish the results for all the world to enjoy.

**Comment by ODEH67** [comment]
@LukeRoss00  @d0x360  guys guys calm down its all just misunderstanding situation, I'm not trying to play the "good guy" here but really nobody talked down to the other or meant to insult. it is just a "Agree/disagree" discussion. each of u have the knowledge specially form an own experience , I hope it is already clear what I'm trying to say (which I meant positively with that) ✌️

**Comment by d0x360** [comment]
I appreciate what you're doing @ODEH67  but the dude was basically calling me stupid for no reason while at the same time seems to be... lacking knowledge on how rendering works.  Different engines yes but they both use the exact same api calls for literally everything aside from function calls to some middleware but that would have zero effect on what we're talking about here.  Apparently making suggestions is a bad thing in this repo.

I also said ask the guy who made the injection tool for some info.  He's already helped other people make the tool work with other games...yah just need to ask.  I handed you the details on how to contact him because apparently you don't understand how google works.

I'm done either way.  I'll take no part in nonsense like this.  If you can't be civil in your own repo then you probably shouldn't have one.  It's not my job to make YOUR mod better

**Comment by AielOlos** [comment]
Hello everybody, 
 First of all, sorry for my bad English. 
 I am trying some configs since 2 days, and I detect something strange. I have a 1080ti and i5 8600k and valve index. Ok, then if I use for example 1800 x 1800 and low settings and 120fps, game see extremly sharp, but gpu is at 100% and game runs to 60 to 80 fps, then if I go to advanced graphic settings in game and resolution scale option, it is set: (in low and medium settings to x2,000 and high setting to x2,500) Ok if you turn off this, performance boost a lot of, but of course, it's too blurry.  Here the magic ... Right now, I start the game 2800x2800 :o, enter in game, and set resolution scale to off. Clarity spectacular, near before, and gpu is 60 -70 of use and 120fps stable. Now I can increase textures, grass and something else more. In some places gpu rise to 95% but all ok. I have some moment where fps was 100 because cpu is bottleneck. In conclusion, for some reason option resolution scale in game is fps kill in comparative to increase resolution. 

Off topic, more or less. If you save options in game or put in config file resolution scale different of x2,000 or x2,500, when you restart the game, you can enter without problem to initial menu, but when load game, 2 secons after game loaded, crash to windows. If you change it in game, play, and change it again before close, all works perfect next time you star game.

**Comment by d0x360** [comment]
Resolution scale affects performance more because it's taking your set
resolution and then raising it to it renders at a higher resolution then
downscales that image to the set resolution and you end up with a cleaner
image with less jaggies.

If you are using 2.25x resolution scaling that means your resolution x 2.25.

I hope that makes sense because I know english isn't your native language...

Basically with rendering scale you are more than doubling whatever your set
resolution is which is why it's eating performance.

Also if your gpu isn't at 90% or higher than either your settings are too
low or your cpu bound.

Best settings would be resolution scale at default and MSAA on 4x.  That
uses less resources than 2.25x super sampling by a significant margin.

**Comment by AielOlos** [comment]
Ok, great! :)

**Comment by Faburden** [comment]
Specs:
RX Vega 64
Ryzen 5 3600
Oculus Quest 2
Mod Version 6

For some reason, my game is running at 5 fps whenever I run it with the mod.
I've reinstalled it tried again, followed all the steps, and no luck.
Inside the headset, it's very laggy and stuttery and it's like a screen following wherever I'm looking.

**Comment by willseee** [comment]
I recently installed your mod, I love how simple you made it. As I am not very good with modification.
Can you recommend best settings so that I have good visual but no screen tear? The M setting from realconfig runs fine, but I would like to try to make it look better. What graphics settings in-game should I try? Should I change my HMD settings in steamVR? Should I change global Nvidia settings? etc... Thank you for your time.I have:

i7-7700K CPU @ 4.20GHz
GTX 1080 Ti
Valve Index
config setting M
GTA 1.0.2189.0
no other mods

**Comment by LukeRoss00** [comment]
@willseee With your 1080 Ti, you should be able to use High settings without problems. If you can tolerate some frame rate drops, you might push the base resolution even further as explained in https://github.com/LukeRoss00/gta5-real-mod#even-more-advanced-resolution-tweaking

**Comment by Just4FunQC** [comment]
Hi! New here and juste iniciate myself to VR experience like 2 week ago and now trying GTA in vr. Fisrt i have to say this game in vr is a really nice and the best project a ever seen. At first i had an black screen but couple of minutes later i was ok for that. Now i'm trying to set the graphic. I experiences lot of blury images. In fact at lunch its super nice but couple of meter further the game start to be laggy like 20 fps.. the vr screen have difficulty to follow my headset. (black aroud the image if i move my headset fast or even slow by time. So for driving car of chopper or plane it's impossible because i see sweet nothing hehe. So i'm here because i know you are expert. I'll give you my setup and tell me what reccommended setting i have to do to get a fluid gaming experience pls.. Notice that french is my usual language so sry for my english ;) 

Mobo: Asus Rog Strix B550-E Gaming wifi
CPU: Ryzen 5 3600 (OC 4ghz)
GPU: XFX Rx 5700XT
RAM: 16GO 3600
Oculus Rift
Latest version on Epic Games GTA V

Note: At first lunch i had in graphic setting 1080x1080. But i think i touched something so now its my monitor resolution i see in game setting. It is ok? Maybe i mess something lol

**Comment by LukeRoss00** [comment]
@Just4FunQC Make sure that GTA V is not running, then run `RealConfig.bat` again. With your card, you should be able to have good frame rate with the High preset. Check that the batch file doesn't give you any errors. After that, do not try to change the resolution from inside the game, because GTA doesn't know about 1:1 square resolutions and will always try to force you back to a canonical resolution that is supported by your monitor.

**Comment by Just4FunQC** [comment]
Thx for your reply @LukeRoss00 ! For the realconfig.bat once it’s open what I have to do to set at high? And how to check my batch file? Thx!

**Comment by Just4FunQC** [comment]
In fact I can’t execute that realconfig. The prompt windows disappear directly

**Comment by LukeRoss00** [comment]
@Just4FunQC Something probably went wrong with your setup. You could try installing the mod again, overwriting old files. Make sure that WinRAR or whatever program you use to extract the mod doesn't give you any errors. Then run `RealConfig.bat` again. It should prompt you to select H, M or L, and it should not exit immediately.

**Comment by Just4FunQC** [comment]
I think my windows is the problem because à can’t do any research in the windows bar.. I’m resetting it. No choice
